src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -323,24 +322,16 @@ public AgentSettings LoadSettings()
 
             string notificationSocketAddress = command.GetNotificationSocketAddress();
 
-            // Get Agent settings
-            var settings = new AgentSettings
-            {
-                AcceptTeeEula = acceptTeeEula,
-                AgentId = agent.Id,
-                AgentName = agentName,
-                NotificationPipeName = notificationPipeName,
-                NotificationSocketAddress = notificationSocketAddress,
-                PoolId = poolId,
-                PoolName = poolName,
-                ServerUrl = serverUrl,
-                WorkFolder = workFolder
-            };
-
-            // This is required in case agent is configured as DeploymentAgent. It will make entry for projectName and DeploymentGroup
-            agentProvider.UpdateAgentSetting(settings);
+            //Add Agent settings
+            agentSettings.AcceptTeeEula = acceptTeeEula;
```

Comment:
most of these should be set either by provider or set inline in the early part of this file.
i think only notificationPipeName, notificationSocketAddress, workFolder should be set here.